app/src/main/java/com/example/dispatcher/presentation/favorites/viewModel/FavoritesViewModel.kt

```diff
@@ -0,0 +1,46 @@
+package com.example.dispatcher.presentation.favorites.viewModel
+
+import android.app.Application
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import com.example.dispatcher.presentation.homepage.model.Article
+import com.example.dispatcher.domain.homepage.repo.ArticleRepository
+
+class FavoritesViewModel(application: Application) : AndroidViewModel(application) {
+
+    private val articleRepository = ArticleRepository(application)
+    private lateinit var articlesList: MutableList<Article>
+
+    private val articlesTitlesLiveData = MutableLiveData<String>()
+
+    init {
+        articlesList = articleRepository.getArticles().toMutableList()
+        updateTitles()
+    }
+
+    fun getFavoritesLiveData(): LiveData<String> {
+        return articlesTitlesLiveData
+    }
+
+    fun addTitle(title: String) {
+        val currentTitles = articlesTitlesLiveData.value ?: ""
+        // Add the new title to the existing titles, separated by a newline
+        val updatedTitles = if (currentTitles.isEmpty()) {
+            title
+        } else {
+            "$currentTitles\n$title"
+        }
+        articlesTitlesLiveData.value = updatedTitles
+    }
+
+    private fun updateTitles() {
+
+        // Keep only the titles of the articles
+        val articlesTitles = articlesList?.joinToString(separator = "\n") { article ->
+            article.title
+        } ?: "No articles found"
```

Comment:
do you know what is the purpose of `?: "No articles found`? is this code reachable?